### ISSUE TITLE
generateKey breaks launchd daemon

### ISSUE THREAD

**Issue by Kidswiss** [Mic92/sops-nix#700]
Hi

If `generateKey` is set, the launchd daemon script will look like this:
```
"/bin/wait4path /nix/store && exec if [[ ! -f '/var/lib/sops-nix/key.txt' ]]; then
  echo generating machine-specific age key...
*snip*
```

https://github.com/Mic92/sops-nix/blob/master/modules/nix-darwin/default.nix#L172
https://github.com/Mic92/sops-nix/blob/master/modules/home-manager/sops.nix#L123

However, the `if` doesn't really work in conjunction with `exec`.

Executing this in bash or zsh will show the issue:
```
exec if [[ ! -f ${escapedAgeKeyFile} ]]; then  
echo test
fi 
zsh: parse error near `then'
```

This breaks the automated mounting of the secrets on startup.

A workaround for now is just setting `generateKey` to false after the first run.

**Comment by mettavi** [comment]
I have exactly the same problem on nix-darwin.

Your solution worked perfectly, thank you so much for posting it.

**Comment by Mic92** [comment]
Is `launchd.daemons.sops-install-secrets` the failing service?

**Comment by mettavi** [comment]
> Is `launchd.daemons.sops-install-secrets` the failing service?

Yes, with `generateKey` set to `true`, `sops-install-secrets` is:

```
"/bin/wait4path /nix/store && exec if [[ ! -f '/var/lib/sops-nix/key.txt' ]]; then
  echo generating machine-specific age key...
*snip*
```
which generates the following error:

```
/bin/sh: -c: line 0: syntax error near unexpected token `then'
/bin/sh: -c: line 0: `/bin/wait4path /nix/store && exec if [[ ! -f /var/lib/sops-nix/key.txt ]]; then'
/bin/sh: -c: line 0: syntax error near unexpected token `then'
/bin/sh: -c: line 0: `/bin/wait4path /nix/store && exec if [[ ! -f /var/lib/sops-nix/key.txt ]]; then'
```

But when `generateKey` is disabled, `sops-install-secrets` is:

```
/bin/sh -c '/bin/wait4path /nix/store && exec 
echo "Setting up secrets..."
*snip*
```
and the error goes away.

**Comment by Kidswiss** [comment]
@Mic92 

The error occurs in the system wide service as well as the user specific one:
* org.nixos.sops-install-secrets-for-users
* org.nixos.sops-install-secrets

**Comment by Mic92** [comment]
Please check the pull request
